services/viva-ms/helpers/populateForm.js

```diff
@@ -0,0 +1,34 @@
+import { generateDataMap, formatAnswer, mergeAnswers } from '../../../libs/formAnswers';
+
+export function populateChildrenAnswers(repeaterInputList, childrenList) {
+  const answers = [];
+
+  childrenList.forEach((child, childrenListIndex) => {
+    for (const [childKey, childValue] of Object.entries(child)) {
+      const childAnswer = repeaterInputList
+        .map(input => {
+          if (input.tags.includes(childKey)) {
+            const inputId = input.id.replace('[*]', childrenListIndex);
+            return formatAnswer(inputId, input.tags, childValue);
+          }
+        })
+        .filter(Boolean);
+
+      answers.push(...childAnswer);
+    }
+  });
+
+  return answers;
+}
+
+export default function populateFormWithVivaChildren(form, formTemplate, vivaChildrenList) {
```

Comment:
This function is very redundant, in terms of what the  populateFormWithPreviousCaseAnswers already does. Is there a way to combine these two functions without some effort. 

I'm aware that the populate code in general is very messy, but making it messier is not a good practice. Better to refactor as we go (y)

Reply:
They do different things. Will keep for now.

Reply:
The code tells a different story, they almost look identical. Except from the handling of childrenList :)
But if it's to much of an effort i will let this of the hook :)